src/Revenue/Migrations/AddPastDonationsToRevenueTable.php

```diff
@@ -63,10 +63,14 @@ public function run() {
 					continue;
 				}
 
+				if ( ! ( $amount = give()->payment_meta->get_meta( $post->ID, '_give_cs_base_amount', true ) ) ) {
+					$amount = give_donation_amount( $post->ID );
+				}
+
 				$revenueData = [
 					'donation_id' => $post->ID,
 					'form_id'     => give_get_payment_form_id( $post->ID ),
-					'amount'      => give_donation_amount( $post->ID ) * 100,
+					'amount'      => $amount * 100,
```

Comment:
You can't make the assumption that all currencies can be multiplied by 100 to get to their smallest unit. You will need to use something like https://github.com/brick/money to handle this.

Reply:
@JasonTheAdams Can you give me a code sample about how can I around donation amount?

Reply:
### Value Objects

| Money   |
| -------------|
| public $amount  |
| public $currency  |
| public static fromArray  |


| Currency  |
| -------------|
| public $symbol  |
| public $currencyPosition  |
| public $decimalSeperator  |
| public $thousandSeperator  |
| public $numberOfDecimals  |


### Helper Function
|  Money  |
| -------------|
| public static getSmallestUnit( Money $money )  |

Currency list: https://github.com/impress-org/givewp/blob/ef833e3671fcddb4e8da5d78bb3a680ecdcd2ca2/includes/currencies-list.php#L23


@JasonTheAdams We already have a list of currency in the core.
What do you think about creating value objects for currency and money and then convert money to the smartest unit?

Reply:
@ravinderk Do you know if GiveWP requires the Intl PHP extension or not?

Reply:
@JasonTheAdams Not now, but can recommend it in future: https://make.wordpress.org/hosting/2019/09/05/re-php-extensions/

Reply:
From our discussion:

```php
$money = Money::of(50.25, 'USD');
$money->getMinorAmount(); // 5025

$money = Money::ofMinor(5025, 'USD');
$money->getAmount(); // 50.25
```

Reply:
Resolved